Observer/Frontend/CustomerLoggedIn.php

```diff
@@ -49,4 +64,9 @@ public function execute(Observer $observer)
             $this->sailthruClient->logger($e);
         }
     }
+    private function shouldUpdate($sid)
+    {
+        $userData = $this->sailthruClient->apiGet('user', [ 'id' => $sid  ]);
+        return $userData['optout_email'] != 'none' || $this->subscriber->getStatus() == Subscriber::STATUS_UNSUBSCRIBED;
```

Comment:
This should be checking optout, but also to see if the newsletter list is missing.

Also, this "or" is always going to eval true if a customer is already unsubscribed in Magento. What is the purpose of checking for STATUS_UNSUBSCRIBED here?

Reply:
i figured it would just be another added buffer of security, but i see your point. i can remove it.